WcaOnRails/app/controllers/competition_tabs_controller.rb

```diff
@@ -27,12 +27,12 @@ def create
 
   def edit
     @competition = competition_from_params
-    @competition_tab = CompetitionTab.find(params[:id])
+    @competition_tab = CompetitionTab.find_by!(id: params[:id], competition: @competition)
```

Comment:
```suggestion
    @competition_tab = @competition.tabs.find(params[:id])
```

Reply:
`find` expects an ID as its only parameter. If you access the tabs through the `has_many :tabs` association in the `Competition` model, you can let Rails handle the DB loading via caches and stuff and don't have to care about manually retrieving the tab yourself.

Reply:
Very nice way of handling it 😎